qt/widgets/plotting/inc/MantidQtWidgets/Plotting/Mpl/ExternalPlotter.h

```diff
@@ -6,30 +6,37 @@
 // SPDX - License - Identifier: GPL - 3.0 +
 #pragma once
 
-#include "DllConfig.h"
-#include "MantidAPI/MatrixWorkspace.h"
+#include "MantidAPI/MatrixWorkspace_fwd.h"
+#include "MantidQtWidgets/Plotting/DllOption.h"
 
 #include <boost/none_t.hpp>
 #include <boost/optional.hpp>
 
+#include <QHash>
+#include <QVariant>
+
 namespace MantidQt {
-namespace CustomInterfaces {
+namespace Widgets {
+namespace MplCpp {
 
 enum MantidAxis { Spectrum, Bin };
 
 /**
  @class ExternalPlotter
  ExternalPlotter is a class used for external plotting within Indirect
  */
-class MANTIDQT_INDIRECT_DLL ExternalPlotter {
+class EXPORT_OPT_MANTIDQT_PLOTTING ExternalPlotter {
 
 public:
   ExternalPlotter();
   virtual ~ExternalPlotter();
 
   virtual void plotSpectra(std::string const &workspaceName, std::string const &workspaceIndices, bool errorBars);
+  virtual void plotSpectra(std::string const &workspaceName, std::string const &workspaceIndices, bool errorBars,
+                           boost::optional<QHash<QString, QVariant>> kwargs);
   virtual void plotCorrespondingSpectra(std::vector<std::string> const &workspaceNames,
-                                        std::vector<int> const &workspaceIndices, bool errorBars);
+                                        std::vector<int> const &workspaceIndices, std::vector<bool> errorBars,
+                                        std::vector<boost::optional<QHash<QString, QVariant>>> kwargs);
```

Comment:
If you create an overload `plotCorrespondingSpectra` without the kwargs (a bit like what you've done for `plotSpectra`) then we can avoid constructing the kwargs in the indirect files